www/js/services/controlHelper.ts

```diff
@@ -0,0 +1,159 @@
+import { DateTime } from 'luxon';
+
+import { getRawEntries } from './commHelper';
+import { logInfo, displayError, logDebug, logWarn } from '../plugin/logger';
+import { FsWindow } from '../types/fileShareTypes';
+import { ServerResponse } from '../types/serverData';
+import i18next from '../i18nextInit';
+
+declare let window: FsWindow;
+
+export const getMyDataHelpers = function (
+  fileName: string,
+  startTimeString: string,
+  endTimeString: string,
+) {
+  const localWriteFile = function (result: ServerResponse<any>) {
+    const resultList = result.phone_data;
+    return new Promise<void>(function (resolve, reject) {
```

Comment:
can this be combined with the similar function that saves `loggerDB` and `userCacheDB`?
Maybe create a new `fileWriterService`?